freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/AuthDistributorService.java

```diff
@@ -0,0 +1,57 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.authdistributor.AuthDistributorProto.UserState;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.authdistributor.GrpcAuthDistributorClient;
+import com.sequenceiq.freeipa.converter.freeipa.user.UmsUsersStateToAuthDistributorUserStateConverter;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UmsUsersState;
+
+@Service
+public class AuthDistributorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthDistributorService.class);
+
+    @Inject
+    private GrpcAuthDistributorClient grpcAuthDistributorClient;
+
+    @Inject
+    private UmsUsersStateToAuthDistributorUserStateConverter umsUsersStateToAuthDistributorUserStateConverter;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public void updateAuthViewForEnvironment(String environmentCrn, UmsUsersState umsUsersState, String accountId) {
+        if (entitlementService.isCdpSaasEnabled(accountId)) {
+            try {
+                LOGGER.debug("Update auth view in auth distributor for env: {}", environmentCrn);
+                UserState userState = umsUsersStateToAuthDistributorUserStateConverter.convert(umsUsersState);
+                grpcAuthDistributorClient.updateAuthViewForEnvironment(environmentCrn, userState);
+                LOGGER.debug("Update auth view in auth distributor finished for env: {}", environmentCrn);
+            } catch (Exception e) {
+                LOGGER.error("Update auth view in auth distributor failed for env: {}", environmentCrn, e);
+            }
+        } else {
+            LOGGER.debug("CDP_SAAS entitlement disabled, skip auth view update for env: {}", environmentCrn);
```

Comment:
do we need this logging? entitlementService is already logging about entitlement check